releases/release-4.0.16.md

```diff
@@ -0,0 +1,109 @@
+---
+title: TiDB 4.0.16 Release Notes
+---
+
+# TiDB 4.0.16 Release Notes
+
+Release Date: December 10, 2021
+
+TiDB version: 4.0.16
+
+## Compatibility changes
+
++ TiKV
+
+    - When casting an invalid utf8 string to real, try to truncate the utf8 prefix instead of returning an error [#11466](https://github.com/tikv/tikv/issues/11466)
+
++ Tools
+
+    + TiCDC
+
+        - Change the default value of Kafka Sink `max-message-bytes` to 1 MB [#2962](https://github.com/pingcap/ticdc/issues/2962)
```

Comment:
```suggestion
        - Change the default value of Kafka Sink `max-message-bytes` to 1 MB to prevent sending too large messages to Kafka clusters [#2962](https://github.com/pingcap/ticdc/issues/2962)
```